File: src/data/script_004_join_data_sets.r
library(dplyr)

message("Reading clickstream data...")
clickstreamDf <- read.csv(file="data/interim/clickstream/clickstream_with_headers.csv", sep=",")
partialClickstreamDf <- clickstreamDf[c("Session_ID", "Product_Object_ID",
                                        "Product_ID", "Request_Date", "Request_Sequence")]

# force certain variables to be categorical:
partialClickstreamDf$Session_ID <- factor(partialClickstreamDf$Session_ID)
partialClickstreamDf$Product_Object_ID <- factor(partialClickstreamDf$Product_Object_ID)
partialClickstreamDf$Request_Sequence <- factor(partialClickstreamDf$Request_Sequence)
summary(partialClickstreamDf)


message("Reading orders data...")
orderDf <- read.csv(file="data/interim/orders/orders_with_headers.csv", sep=",")
partialOrdersDf <- orderDf[c("Order_Session_ID", "Order_Line_Session_ID",
                             "Product_Object_ID", "Product_ID")]

# force certain variables to be categorical:
partialOrdersDf$Order_Session_ID <- factor(partialOrdersDf$Order_Session_ID)
partialOrdersDf$Product_Object_ID <- factor(partialOrdersDf$Product_Object_ID)
#Order_Line_Session_ID <- factor(Order_Line_Session_ID)
summary(partialOrdersDf)


# join data:
message("\n\n\nJoining data...")
joinedDf <- partialClickstreamDf %>%
  inner_join(partialOrdersDf, by=c('Session_ID'='Order_Line_Session_ID'
#                                                                   'Session_ID'='Order_Session_ID',
#                                                                   'Product_ID'='Product_ID',
                                                                   ))
message("Done!")

message("\n\n\nSUMMARY")
summary(joinedDf)

message("HEAD")
head(joinedDf)

message("Writing the joined data frame to disc...")
#write.csv(joinedDf, file="data/interim/joined/joined_df.csv", row.names = TRUE)
message("Done!")

File: src/data/script_004_join_data_sets.py
import pandas as pd
import os
import os.path

def merge_files(clickstream_file_path: str, orders_file_path: str) -> pd.DataFrame:
    """ Cleans a given CSV file from NaN values etc. and returns the data frame. """
    print('Loading file ' + clickstream_file_path)
    if os.path.isfile(clickstream_file_path):
        clickstream_df = pd.read_csv(clickstream_file_path, encoding = 'latin-1')
        print('Loading file ' + orders_file_path)
        if os.path.isfile(orders_file_path):
            orders_df = pd.read_csv(orders_file_path, encoding = 'latin-1')

            # force certain variables to be categorical:
            clickstream_df['Session_ID'] = clickstream_df['Session_ID'].astype('category')
            orders_df['Order_Session_ID'] = orders_df['Order_Session_ID'].astype('category')

            # join data:
            return pd.merge(clickstream_df, orders_df, how = 'inner', left_on='Session_ID', right_on='Order_Session_ID')
        else:
            print('The data file does not exist!')
    else:
        print('The data file does not exist!')

print('Merging data files...')
joined_df = merge_files(r'data/interim/clickstream/clickstream_cleaned_py.csv', r'data/interim/orders/orders_cleaned_py.csv')
print('joined_df.describe()...')
print(joined_df.describe(include = 'all'))
